src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceSerializer.cs

```diff
@@ -505,5 +808,214 @@ private void WriteOperation(ODataOperation operation)
 
             this.JsonLightOutputContext.JsonWriter.EndObjectScope();
         }
+
+        /// <summary>
+        /// Asynchronously writes an association link property annotation.
+        /// </summary>
+        /// <param name="propertyName">The name of the navigation property for which to write the association link.</param>
+        /// <param name="associationLinkUrl">The association link URL to write.</param>
+        private async Task WriteAssociationLinkAsync(string propertyName, Uri associationLinkUrl)
+        {
+            Debug.Assert(!String.IsNullOrEmpty(propertyName), "!string.IsNullOrEmpty(propertyName)");
+            Debug.Assert(associationLinkUrl != null, "associationLinkUrl != null");
+
+            // The association link URL is a property annotation "NestedResourceInfoName@odata.associationLinkUrl: 'url'"
+            await this.AsynchronousODataAnnotationWriter.WritePropertyAnnotationNameAsync(propertyName, ODataAnnotationNames.ODataAssociationLinkUrl)
+                .ConfigureAwait(false);
+            await this.AsynchronousJsonWriter.WriteValueAsync(this.UriToString(associationLinkUrl))
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Asynchronously writes a group of operation (all actions or all functions) that have the same "metadata".
+        /// </summary>
+        /// <remarks>
+        /// Expects the actions or functions scope to already be open.
+        /// </remarks>
+        /// <param name="operations">A grouping of operations that are all actions or all functions and share the same "metadata".</param>
+        private async Task WriteOperationMetadataGroupAsync(IGrouping<string, ODataOperation> operations)
+        {
+            this.ValidateOperationMetadataGroup(operations);
+            await this.JsonLightOutputContext.AsynchronousJsonWriter.WriteNameAsync(operations.Key)
+                .ConfigureAwait(false);
+            bool useArray = operations.Count() > 1;

```

Comment:
Would `Any()` ? be better hear?

Reply:
@habbes @marabooy Notice that here the check is for collection having **more than** one item. `Any()` is equivalent to `Count() > 0`